### DISCUSSION TITLE
How to intergrate github events with discord?

### DISCUSSION THREAD

**Discussion by popstar7** [Finalgoal231/finalGoal#107] [About github] [answered]
Please explain about webhooks

**Comment by noahstoner** [comment, marked as answer]
Hello.
There is the following link.
https://gist.github.com/jagrosh/5b1761213e33fc5b54ec7f6379034a22

With this discord webhooks, we can make any discord bot.

**Reply by popstar7** [reply]
Great answer.

**Comment by sergiourrego** [comment]
To integrate GitHub events (such as push, pull requests, etc.) with a Discord server, you can use GitHub Webhooks and set them to send events to a Discord channel. Here's a step-by-step guide on how to do it:

1. Create a Discord Webhook
First, you need to create a webhook in your Discord server to receive GitHub event notifications.

Open Discord and navigate to your server.
Click the gear icon next to the channel where you want to receive the GitHub events.
Under the channel settings, go to Integrations.
Click Create Webhook.
Name the webhook (e.g., "GitHub Webhook") and copy the Webhook URL (you will need this later).
Click Save Changes.
2. Set Up a GitHub Webhook
Navigate to the GitHub repository where you want to track events.
Go to the repository's Settings (located in the top bar of your repo page).
On the left sidebar, click on Webhooks.
Click the Add webhook button.
In the Payload URL field, paste the Discord Webhook URL you copied earlier.
Set the Content type to application/json.
Choose the events you want to send to Discord (e.g., push events, pull requests, issues, etc.). You can also select Let me select individual events to choose specific ones or Send me everything to send all events.
Click Add webhook to complete the process.
3. Customize Webhook Payload (Optional)
You can customize the JSON payload format or add filtering logic if needed using services like Zapier, IFTTT, or writing a custom server using Node.js that sits between GitHub and Discord.

Example of the Events on Discord
Once the webhook is set up, when someone pushes a commit, opens a pull request, or triggers any other event in the GitHub repository, Discord will automatically receive and display the notification in the specified channel.

The messages will typically include:

Event type (push, pull request, etc.).
Commit messages.
Links to the repository or the specific event.
Information about the user who triggered the event.
Example of a Push Event Notification:
When a push is made to the repository, you’ll see a message like:

css
Copy code
[Repository Name]
UserName pushed 3 commits to branch main
- Commit 1 message
- Commit 2 message
- Commit 3 message
This is a quick and easy way to stay informed about GitHub activity in your Discord server.

If you need to extend or fine-tune this integration further (e.g., filtering events, formatting messages), I can guide you through that as well.
